prow/cmd/mkpj/main.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/ghodss/yaml"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/kube"
+	"k8s.io/test-infra/prow/plank"
+)
+
+var (
+	jobName    = flag.String("job", "", "Job to run.")
+	configPath = flag.String("config-path", "", "Path to config.yaml.")
+)
+
+func main() {
+	flag.Parse()
+
+	if *jobName == "" {
+		logrus.Fatal("Must specify --job.")
+	}
+
+	conf, err := config.Load(*configPath)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error loading config.")
+	}
+
+	var pjs kube.ProwJobSpec
+	var found bool
+	var needsBaseRef bool
+	var needsPR bool
+	for fullRepoName, ps := range conf.Presubmits {
+		org, repo, err := splitRepoName(fullRepoName)
+		if err != nil {
+			logrus.WithError(err).Fatal("Invalid repo name.")
+		}
+		for _, p := range ps {
+			if p.Name == *jobName {
+				pjs = plank.PresubmitSpec(p, kube.Refs{
+					Org:   org,
+					Repo:  repo,
+					Pulls: []kube.Pull{{}},
+				})
+				found = true
+				needsBaseRef = true
+				needsPR = true
+			}
+		}
+	}
+	for fullRepoName, ps := range conf.Postsubmits {
+		org, repo, err := splitRepoName(fullRepoName)
+		if err != nil {
+			logrus.WithError(err).Fatal("Invalid repo name.")
+		}
+		for _, p := range ps {
+			if p.Name == *jobName {
+				pjs = plank.PostsubmitSpec(p, kube.Refs{
+					Org:  org,
+					Repo: repo,
+				})
+				found = true
+				needsBaseRef = true
+			}
+		}
+	}
+	for _, p := range conf.Periodics {
+		if p.Name == *jobName {
+			pjs = plank.PeriodicSpec(p)
```

Comment:
It picks periodic > postsubmit > presubmit. We could add an extra flag that says "use this one" if we'd like.